src/parser/Tokenizer.cc

```diff
@@ -264,7 +264,6 @@ Parser::Tokenizer::int64(int64_t & result, int base, bool allowSign, const SBuf:
     if (base == 0) {
         if ( *s == '0') {
             base = 8;
-            ++s;
```

Comment:
We do not need to advance input here because our integer value computing math below should handle any leading zeros correctly. Advancing input leads to the bug this PR is fixing.

An alternative fix is to treat "0" and similar inputs specially. The fix proposed by this PR feels like a slightly lesser evil to me.

P.S. This comment is not requesting any changes.

Reply:
IIRC, this was for correct auto detection of octal "0" and "00" cases.